### ISSUE TITLE
Dashboard logs disappear when returning to Dash

### ISSUE THREAD

**Issue by smunteankestra** [kestra-io/kestra#5251]
### Describe the issue

Preconditions: have any logs in place in dashboard
Step 1: Check logs on Dashboard
Step 2: Navigate to any other page for example "logs"
Step 3: Go back to Dashboard

Expected: logs shown
Actual: no logs - a refresh will fix it

https://github.com/user-attachments/assets/a0c0bc10-e27a-4874-a8b3-1c142166471d

### Environment

- Kestra Version: develop
- Operating System (OS/Docker/Kubernetes):

**Comment by Suleman9580** [comment]
Is the issue solved ?
If not, can it contribute..

**Comment by MilosPaunovic** [comment]
Sure @Suleman9580, go for it! :rocket:

**Comment by mithindev** [comment]
Hello @MilosPaunovic ,

Thank you so much for your time! I really value your time and truly appreciate it. Although I'm not assigned to this issue, I was curious and decided to go through it. I noticed the following:

- When the page is refreshed (or loads from scratch), the props.data for the plot has 31 elements. However, when I navigate to another page and then return, the last element is somehow deleted. I'm currently working on figuring out why this happens.

-- Just after refreshing (before navigating away and coming back):
<img width="700" alt="image" src="https://github.com/user-attachments/assets/37541ee2-4ce1-4742-962f-7f59b9af156a">

-- After navigating to another route and coming back:
<img width="693" alt="image" src="https://github.com/user-attachments/assets/dc99e0a0-a108-4fea-abdf-8fff943215f9">

One brute-force solution could be to trigger a refresh for just the Plot component if the length has changed, but I was figuring out the cause of this deletion first.

Thank you once again for your time! Please let me know if I’m misunderstanding anything or if I’m on the wrong track. 😊

**Comment by MilosPaunovic** [comment]
I am not sure I quite understand your proposed solution @mithindev, but as nobody submitted PR yet, you are free to do it.

**Comment by Danesh-md** [comment]
is this issue fixed or someone working on it?

**Comment by MilosPaunovic** [comment]
@mithindev are you working on this?

**Comment by mithindev** [comment]
yes @MilosPaunovic I am! I just need a couple of days, to fix it.

**Comment by mithindev** [comment]
@Danesh-md , we can work on this issue together, if you want. 😀

**Comment by Danesh-md** [comment]
done bro

**Comment by Danesh-md** [comment]
@mithindev sure bro. can you share ur progress

**Comment by MilosPaunovic** [comment]
I'd enjoy to see Kestra gathering multiple developers from the community and them joining forces to work on the same problem, helping each other! Awesome initiative, guys @mithindev @Danesh-md! 🚀

**Comment by mithindev** [comment]
Hello @Danesh-md,

I’ve been investigating this issue for a few days, exploring various potential causes. Initially, I suspected state management, as the issue only occurred when navigating between pages but worked after a refresh. This led me down a different path. (Learned a lot along the way! 😊)

The root cause was in the logic used to filter dates, specifically how the time range was calculated. In the handleDatesUpdate method, if startDate and endDate are provided, the filters update correctly. However, for a “30 days” timeRange, it was unclear if the current day was included.

Fix:
I updated the date logic:
- startDate: Subtracted 30 days from today using moment().subtract(30, 'days').startOf('day') to ensure the correct range.
- endDate: Added tomorrow’s end of day with moment().add(1, 'days').endOf('day') to capture today’s data and provide a buffer for future logs.

After this adjustment, the filters worked as expected, resolving the issue.

**Comment by tchiotludo** [comment]
no more reproduction on develop
